pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4251,7 +4243,7 @@ private RestException topicNotFoundReason(TopicName topicName) {
                     == null ? "has no metadata" : "has zero partitions";
             return new RestException(Status.NOT_FOUND, String.format(
                     "Partitioned Topic not found: %s %s", topicName.toString(), topicErrorType));
-        } else if (!internalGetList(Optional.empty()).contains(topicName.toString())) {
+        } else if (!internalGetListAsync(Optional.empty()).join().contains(topicName.toString())) {
```

Comment:
We need to avoid to using the `join()` and `get()`.

Reply:
Should we keep  both `internalGetList ` and  `internalGetListAsync`? @nodece

Reply:
It looks like you need to make the `topicNotFoundReason` to async.

Reply:
There has existed a method `topicNotFoundReasonAsync`,  but `topicNotFoundReason` are still invoked by `getTopicReference`, which appears in many methods. I think it's better to cleanup `topicNotFoundReason` when other methods have replaced `getTopicReference` with `getTopicReferenceAsync`.So  it's looks OK using  `join` here @nodece

Reply:
Ok, it's better using `get(pulsar().getConfiguration().getMetadataStoreOperationTimeoutSeconds(), TimeUnit.SECONDS)` instead of `join()`.

Reply:
sync(() -> topicNotFoundReasonAsync()